integration_tests/testhelpers/stdin_simulator.go

```diff
@@ -0,0 +1,72 @@
+package testhelpers
+
+import (
+	"io"
+	"os"
+)
+
+// StdinSimulator provides a clean abstraction for simulating stdin input in tests.
+// It handles all the complex pipe management and cleanup automatically.
+//
+// Example usage:
+//   simulator, err := NewStdinSimulator("m\nd\nq\n")
```

Comment:
Error: integration_tests/testhelpers/stdin_simulator.go:12:1: File is not properly formatted (gofmt)
  //   simulator, err := NewStdinSimulator("m\nd\nq\n")
  ^
  
  Error: issues found